payjoin/src/uri/error.rs

```diff
@@ -7,10 +11,25 @@ pub(crate) enum InternalPjParseError {
     DuplicateParams(&'static str),
     MissingEndpoint,
     NotUtf8,
-    BadEndpoint,
+    BadEndpoint(BadEndpointError),
     UnsecureEndpoint,
 }
 
+#[derive(Debug)]
+pub enum BadEndpointError {
+    UrlParse(ParseError),
+    FragmentParse(ParseEndpointFragmentError),
```

Comment:
Rather than import at the top level, I think we want to feature gate this line (and the line in the Display impl) and fully qualify the type, if we even need multiple variants in the FragmentParse variant.

```suggestion
    #[cfg(feature = "v2")]
    FragmentParse(super::url_ext::ParseEndpointFragmentError),
```